File: cslavb/Csla/Windows/ObjectSaveException.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Windows

  ''' <summary>
  ''' Exception indicating a failure during an object
  ''' save operation.
  ''' </summary>
  Public Class ObjectSaveException

    Inherits Exception

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    Public Sub New()
      MyBase.New()
    End Sub

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="message">
    ''' Exception message text.
    ''' </param>
    Public Sub New(ByVal message As String)
      MyBase.New(message)
    End Sub

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="message">
    ''' Exception message text.
    ''' </param>
    ''' <param name="innerException">
    ''' Reference to an inner exception.
    ''' </param>
    Public Sub New(ByVal message As String, ByVal innerException As Exception)
      MyBase.New(message, innerException)
    End Sub

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="innerException">
    ''' Reference to an inner exception.
    ''' </param>
    Public Sub New(ByVal innerException As Exception)
      MyBase.New(My.Resources.ExceptionOccurredDuringSaveOperation, innerException)
    End Sub

  End Class

End Namespace

File: Source/Csla.Windows.Shared/ObjectSaveException.cs
//-----------------------------------------------------------------------
// <copyright file="ObjectSaveException.cs" company="Marimer LLC">
//     Copyright (c) Marimer LLC. All rights reserved.
//     Website: https://cslanet.com
// </copyright>
// <summary>Exception indicating a failure during an object</summary>
//-----------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Csla.Properties;

namespace Csla.Windows
{
  /// <summary>
  /// Exception indicating a failure during an object
  /// save operation.
  /// </summary>
  public class ObjectSaveException : Exception 
  {
    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    public ObjectSaveException() : base() { }

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="message">
    /// Exception message text.
    /// </param>
    public ObjectSaveException(string message) : base(message) { }

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="message">
    /// Exception message text.
    /// </param>
    /// <param name="innerException">
    /// Reference to an inner exception.
    /// </param>
    public ObjectSaveException(string message, Exception innerException) : base(message, innerException) { }

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="innerException">
    /// Reference to an inner exception.
    /// </param>
    public ObjectSaveException(Exception innerException) : base(Resources.ExceptionOccurredDuringSaveOperation, innerException) { }
  }
}